### DISCUSSION TITLE
Jetstream backoff

### DISCUSSION THREAD

**Discussion by alberk8** [nats-io/nats.net#421] [Q&A] [answered]
I have set the jetstream backoff, but the redelivery/backoff does not seems to work.  I mean I expect a redelivery based on the backoff. 
The ConsumeAsync just waits forever until a new message is published.

One other question, when the maxDeliver is up, what happens to the undelivered message?

```
var consumer = await jsContext.CreateOrUpdateConsumerAsync("EVENTS", new ConsumerConfig(consumerName)
{
    DeliverPolicy = ConsumerConfigDeliverPolicy.All,
    FilterSubject = "data.>",
    AckPolicy = ackPolicy,
    AckWait = ackWait,
    
    MaxDeliver = 5,
    Backoff = new List<long>() { (long) TimeSpan.FromSeconds(1).TotalSeconds, (long)TimeSpan.FromSeconds(2).TotalSeconds }
});

 await foreach (NatsJSMsg<Person> msg in consumer.ConsumeAsync<Person>())
 {
     Console.WriteLine($"Subject: {msg.Subject} Size: {msg.Size}");
     var md = msg.Metadata;
     if (md.HasValue)
     {
        Console.WriteLine($"Num Delivered: {md.Value.NumDelivered} TimeStamp: {md.Value.Timestamp.ToLocalTime()}");
        Console.WriteLine($"Consumer Seq: {md.Value.Sequence.Consumer} Stream Seq: {md.Value.Sequence.Stream}");
     }
     await msg.NakAsync();
 }
```

**Comment by mtmk** [comment, marked as answer]
AFAIK after maxDeliver times of redelivery, that message would be marked as delivered or terminated (not sure of the actual state) but basically should not be redelivered for that consumer again.

as for back-off, i think that should be in nanoseconds:
* see also #422
